docs/Makefile

```diff
@@ -7,14 +7,43 @@ SPHINXOPTS    ?= -q
 SPHINXBUILD   ?= sphinx-build
 SOURCEDIR     = .
 BUILDDIR      = _build
+VENV          = env/bin/activate
+PORT = 8080
 
 # Put it first so that "make" without argument is like "make help".
 help:
 	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
 
+install:
+	@echo "... setting up virtualenv"
+	python3 -m venv env
+	. $(VENV); pip install --upgrade -r ../requirements-doc.txt -r ../requirements-focal.txt
```

Comment:
what about the requirements in `.sphinx/pinned-requirements.txt`? It's curious that the docs can build without them

Reply:
It's also because i didn't add all the extensions, like sphinx_tabs or didn't upgrade the version of the furo theme, for example, but I can add the extensions if you want in the requirements-doc.txt file and conf.py.

Reply:
can we use only the `.sphinx/pinned-requirements.txt` file?

Reply:
no because :

- Need to solve some version issues because some need sphinx>6 and others <4
- We will need to add our dependencies

For theses reasons, I think it's better to use requirements-doc

Reply:
I see. We should probably take some time to clean-up those dependencies (iirc rockcraft doesn't need autodocs because we create docs from the code) but we can do that in the future, can you please create an issue for it?

Reply:
Here the issue: https://github.com/canonical/rockcraft/issues/306